src/main/java/io/confluent/connect/elasticsearch/bulk/BulkProcessor.java

```diff
@@ -77,7 +77,7 @@
   // shared state, synchronized on (this), may be part of wait() conditions so need notifyAll() on
   // changes
   private final Deque<R> unsentRecords;
-  private final ConcurrentHashMap<R, SinkRecord> recordMap;
+  protected final ConcurrentHashMap<R, SinkRecord> recordMap; // visible for tests
```

Comment:
can we change the name of this variable to something more intuitive?
Also, let's not bake the type in the variable name. 
`recordsToReportOnError` ?